terminal/runereader.go

```diff
@@ -31,7 +31,13 @@ func (rr *RuneReader) printChar(char rune, mask rune) {
 	}
 }
 
-func (rr *RuneReader) ReadLine(mask rune) ([]rune, error) {
+type OnRuneFn func(rune, []rune) ([]rune, bool, error)
+
+func (rr *RuneReader) ReadLine(mask rune, onRunes ...OnRuneFn) ([]rune, error) {
+	return rr.ReadLineWithDefault(mask, []rune{}, onRunes...)
+}
+
+func (rr *RuneReader) ReadLineWithDefault(mask rune, d []rune, onRunes ...OnRuneFn) ([]rune, error) {
```

Comment:
I'm a bit nervous about adding this much logic to a core terminal method. Can you walk me through why the current implementation doesn't work?

Reply:
the current input with the suggest allows you to switch between selecting in the list and typing it normally.

if you are typing and press tab, the list appears and works like a select, but if you type a letter it goes back to a normal input.

with the previous implementation using ReadRune directly we did just re-print the "input format" and started printing each rune as it was typed.

the function ReadLine does not allow such control... but if we can set a default input, but keeping the cursor knowing that is a editable section then we can set the input for the user.

it is working like this: https://asciinema.org/a/NjD9zy242lY25NXUreS9J2Lim

```go
package main

import (
	"fmt"
	"os"

	"github.com/AlecAivazis/survey/v2/terminal"
)

func main() {
	rr := terminal.NewRuneReader(terminal.Stdio{
		In:  os.Stdin,
		Out: os.Stdin,
		Err: os.Stderr,
	})
	fmt.Print("some input: ")
	rr.SetTermMode()
	line, err := rr.ReadLineWithDefault(0, []rune("just a start"))

	fmt.Printf("Line: %s, Err: %s", string(line), err)
}
```

i did that way because you changed to use [`ReadLine`](https://github.com/AlecAivazis/survey/pull/361/commits/b52624ad8689c00aeb854fa9469fbf07c8f090e7), I think this is better than adding more ifs for the arrows and duplicating code for the same effect.

Reply:
i think it does not support it at least.

Reply:
Ahhh okay so basically the default argument is a way to keep the input in place between tabs - got it! Thanks for explaining